### DISCUSSION TITLE
How can I get blockInfo by txId

### DISCUSSION THREAD

**Discussion by artz001** [hyperledger/fabric-gateway#504] [Q&A] [answered]
I can't find a similar way like fabric-gateway-java : 
 

Channel channel = network.getChannel();
 channel.queryBlockByTransactionID("");

**Comment by bestbeforetoday** [comment, marked as answer]
You can evaluate the **GetBlockByTxID** transaction function on the **qscc** system chaincode. The qscc system chaincode is not well documented but the implementation is here:

https://github.com/hyperledger/fabric/blob/main/core/scc/qscc/query.go

GetBlockByTxID takes two arguments: channel name and transaction ID. It returns a serialized [Block](https://hyperledger.github.io/fabric-protos/protos.html#common-Block) protocol buffer message. You can deserialize this using the [fabric-protos](https://hyperledger.github.io/fabric-protos/) language bindings for the Fabric protocol buffer messages:

```java
var block = org.hyperledger.fabric.protos.common.Block.parseFrom(responseBytes);
```

The [off_chain_data](https://github.com/hyperledger/fabric-samples/tree/main/off_chain_data) sample demonstrates how to navigate the block structure and retrieve information. You might want to reuse some or all of the [parser](https://github.com/hyperledger/fabric-samples/tree/main/off_chain_data/application-java/app/src/main/java/parser) code from there.
